src/main/java/com/celements/web/plugin/cmd/FileBaseTagsCmd.java

```diff
@@ -23,42 +23,88 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.xwiki.context.Execution;
 import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.model.reference.WikiReference;
 
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.celements.model.access.exception.DocumentSaveException;
+import com.celements.navigation.INavigationClassConfig;
 import com.celements.navigation.TreeNode;
 import com.celements.navigation.filter.InternalRightsFilter;
 import com.celements.navigation.service.ITreeNodeService;
 import com.xpn.xwiki.XWikiContext;
-import com.xpn.xwiki.XWikiException;
 import com.xpn.xwiki.doc.XWikiDocument;
 import com.xpn.xwiki.objects.BaseObject;
 import com.xpn.xwiki.web.Utils;
 
 public class FileBaseTagsCmd {
 
-  private final static Logger Logger = LoggerFactory.getLogger(FileBaseTagsCmd.class);
+  private final static Logger LOGGER = LoggerFactory.getLogger(FileBaseTagsCmd.class);
 
-  public static final String FILEBASE_TAG_CLASS = "Classes.FilebaseTag";
-  private ITreeNodeService treeNodeSrv = Utils.getComponent(ITreeNodeService.class);
+  public final static String FILEBASE_TAG_CLASS = "Classes.FilebaseTag";
+  private final ITreeNodeService treeNodeSrv = Utils.getComponent(ITreeNodeService.class);
+  private final IModelAccessFacade modelAccess = Utils.getComponent(IModelAccessFacade.class);
+  private final INavigationClassConfig navClassConfig = Utils.getComponent(
+      INavigationClassConfig.class);
+  private final Execution execution = Utils.getComponent(Execution.class);
 
+  private XWikiContext getContext() {
+    return (XWikiContext) execution.getContext().getProperty("xwikicontext");
+  }
+
+  /**
+   * @deprecated since 1.141 instead use getAllFileBaseTags()
+   */
+  @Deprecated
   public List<TreeNode> getAllFileBaseTags(XWikiContext context) {
-    return treeNodeSrv.getSubNodesForParent("", getTagSpaceName(context),
-        new InternalRightsFilter());
+    return getAllFileBaseTags();
   }
 
+  public List<TreeNode> getAllFileBaseTags() {
+    return treeNodeSrv.getSubNodesForParent(getTagSpaceRef(), new InternalRightsFilter());
+  }
+
+  /**
+   * @deprecated since 1.141 instead use getTagSpaceRef()
+   */
+  @Deprecated
   public String getTagSpaceName(XWikiContext context) {
-    String centralFileBaseName = getCentralFileBaseFullName(context) + ".";
-    return centralFileBaseName.substring(0, centralFileBaseName.indexOf('.'));
+    return getTagSpaceRef().getName();
   }
 
+  public SpaceReference getTagSpaceRef() {
+    String centralFileBaseName = getCentralFileBaseFullName() + ".";
+    String spaceName = centralFileBaseName.substring(0, centralFileBaseName.indexOf('.'));
+    return new SpaceReference(spaceName, new WikiReference(getContext().getDatabase()));
```

Comment:
this statements throws an IAE if getCentralFileBaseFullName() returns the default empty string